File: src/main/scala/jk_5/nailed/server/mappack/metadata/xml/XmlTutorialStage.scala
/*
 * Nailed, a Minecraft PvP server framework
 * Copyright (C) jk-5 <http://github.com/jk-5/>
 * Copyright (C) Nailed team and contributors <http://github.com/nailed/>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the MIT License.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
 * for more details.
 *
 * You should have received a copy of the MIT License along with
 * this program. If not, see <http://opensource.org/licenses/MIT/>.
 */

package jk_5.nailed.server.mappack.metadata.xml

import jk_5.nailed.api.mappack.MappackConfigurationException
import jk_5.nailed.api.mappack.tutorial.TutorialStage
import org.jdom2.Element

import scala.collection.convert.wrapAsScala._

/**
 * No description given
 *
 * @author jk-5
 */
class XmlTutorialStage(element: Element) extends TutorialStage {
  if(element.getChild("title", element.getNamespace) == null) throw new MappackConfigurationException("Tutorial stage does not have a title")

  override val title = element.getChild("title", element.getNamespace).getText
  override val messages = if(element.getChild("messages", element.getNamespace) == null) new Array[String](0) else element.getChild("messages", element.getNamespace).getChildren.map(_.getText).toArray
  override val teleport = if(element.getChild("teleport", element.getNamespace) == null) null else XmlMappackWorld.readLocation(element.getChild("teleport", element.getNamespace))
}

File: src/main/java/jk_5/nailed/server/mappack/metadata/xml/XmlTutorialStage.java
package jk_5.nailed.server.mappack.metadata.xml;

import jk_5.nailed.api.mappack.MappackConfigurationException;
import jk_5.nailed.api.mappack.metadata.TutorialStage;
import jk_5.nailed.api.util.Location;
import org.jdom2.Element;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.util.ArrayList;
import java.util.List;

public class XmlTutorialStage implements TutorialStage {

    private final String title;
    private final String[] messages;
    private final Location teleport;

    public XmlTutorialStage(Element element) throws MappackConfigurationException {
        this.title = XmlUtils.getRequiredText(element, "title");

        if(element.getChild("messages", element.getNamespace()) == null){
            this.messages = new String[0];
        }else{
            List<String> messages = new ArrayList<String>();
            for(Element e : element.getChild("messages", element.getNamespace()).getChildren()){
                messages.add(e.getText());
            }
            this.messages = messages.toArray(new String[messages.size()]);
        }

        if(element.getChild("teleport", element.getNamespace()) == null){
            this.teleport = null;
        }else{
            this.teleport = XmlMappackWorld.readLocation(element.getChild("teleport", element.getNamespace()));
        }
    }

    @Nonnull
    @Override
    public String title() {
        return this.title;
    }

    @Nonnull
    @Override
    public String[] messages() {
        return this.messages;
    }

    @Nullable
    @Override
    public Location teleport() {
        return this.teleport;
    }
}
